tslint/relayOperationGenericsRule.js

```diff
@@ -0,0 +1,277 @@
+// @ts-check
+
+const Lint = require("tslint")
+const ts = require("typescript")
+const { BREAK, parse, visit } = require("graphql")
+
+class Rule extends Lint.Rules.AbstractRule {
+  /**
+   * @param {ts.SourceFile} sourceFile
+   */
+  apply(sourceFile) {
+    return this.applyWithWalker(
+      new RelayOperationGenericsWalker(sourceFile, this.getOptions())
+    )
+  }
+}
+
+class RelayOperationGenericsWalker extends Lint.RuleWalker {
+  constructor(sourceFile, options) {
+    super(sourceFile, options)
+    this._imports = []
+  }
+
+  /**
+   * @return {ts.ImportDeclaration[]}
+   */
+  getImports() {
+    return this._imports
+  }
+
+  /**
+   *
+   * @param {ts.ImportDeclaration} node
+   */
+  visitImportDeclaration(node) {
+    this._imports.push(node)
+    super.visitImportDeclaration(node)
+  }
+
+  /**
+   * @param {ts.JsxSelfClosingElement} node
+   */
+  visitJsxSelfClosingElement(node) {
+    // TODO: So many hoops to jump through without TS :/
+    /** @type {any} */
+    let any
+
+    if (node.tagName.getText() === "QueryRenderer") {
+      for (const property of node.attributes.properties) {
+        if (
+          property.kind === ts.SyntaxKind.JsxAttribute &&
+          property.name.getText() === "query"
+        ) {
+          const initializer = property.initializer
+          if (initializer.kind === ts.SyntaxKind.JsxExpression) {
+            this.visitOperationConfiguration(
+              node,
+              initializer.expression,
+              node.tagName
+            )
+          } else {
+            this.addFailureAtNode(
+              initializer,
+              "expected a graphql`…` tagged-template expression"
+            )
+          }
+          break
+        }
+      }
+    }
+
+    super.visitJsxSelfClosingElement(node)
+  }
+
+  /**
+   * @param {ts.CallExpression} node
+   */
+  visitCallExpression(node) {
+    // TODO: So many hoops to jump through without TS :/
+    /** @type {any} */
+    let any
+
+    any = node.expression
+    /** @type {ts.Identifier} */
+    const functionName = any
+    if (functionName.text === "commitMutation") {
+      any = node.arguments[1]
+      /** @type {undefined | ts.ObjectLiteralExpression} */
+      const config = any
+      if (config && config.kind === ts.SyntaxKind.ObjectLiteralExpression) {
+        // any = this.visitOperationConfiguration(node, config, functionName)
```

Comment:
Lil comment, but can remove in follow-up

Reply:
Good catch! Yup, will do in follow-up